dashboard/src/components/AppList/AppList.tsx

```diff
@@ -20,7 +20,7 @@ import "./AppList.css";
 import AppListGrid from "./AppListGrid";
 
 function AppList() {
-  const location = useLocation();
+  const location = ReactRouter.useLocation();
```

Comment:
Just out of curiosity: why this change?

Reply:
that's an interesting "gotcha". The problem is how `useLocation` is exported on it's package. It's directly exported as a function:

```
export function useLocation<S = H.LocationState>(): H.Location<S>;
```

but for mocking it as a spy, we are importing the whole `ReactRouter`: `jest.spyOn(ReactRouter, "useLocation")`. The problem is that if in the functional file, we only import the function (`import { useLocation } from react-router`) it won't use the mocked version of `ReactRouter` and  will import the real function. (testing things).

The thing is that this doesn't happen with other methods (like `useState` that I'm also using here) because it's exported as a method of a namespace:

```
declare namespace React {
...
    function useState<S>(initialState: S | (() => S)): [S, Dispatch<SetStateAction<S>>];
```

so we will be always mocking the namespace, regardless how you import it. :)

Reply:
Ahhhh, got it. Thanks a lot for the explanation! It will be useful for my future tests :P